Source/OIDServiceDiscovery.m

```diff
@@ -191,7 +195,12 @@ + (BOOL)supportsSecureCoding {
 
 - (nullable instancetype)initWithCoder:(NSCoder *)aDecoder {
   NSError *error;
-  NSDictionary *dictionary = [[NSDictionary alloc] initWithCoder:aDecoder];
+  NSDictionary *dictionary = nil;
+  if (@available(iOS 11.0, *)) {
```

Comment:
I think it is safe to replace the old encoding/decoding implementation for every versions of iOS.